lib/spack/spack/compilers/clang.py

```diff
@@ -29,22 +48,29 @@ class Clang(Compiler):
     # Subclasses use possible names of Fortran 90 compiler
     fc_names = ['flang', 'gfortran', 'xlf90_r']
 
-    # Named wrapper links within lib/spack/env
-    link_paths = {'cc': 'clang/clang',
-                  'cxx': 'clang/clang++'}
-
-    if sys.platform == 'darwin':
-        # Use default wrappers for fortran, in case provided in
-        # compilers.yaml
-        link_paths['f77'] = 'clang/gfortran'
-        link_paths['fc'] = 'clang/gfortran'
-    elif spack.architecture.sys_type() == 'linux-rhel7-ppc64le':
-        # This platform uses clang with IBM XL Fortran compiler
-        link_paths['f77'] = 'xl_r/xlf_r'
-        link_paths['fc'] = 'xl_r/xlf90_r'
-    else:
-        link_paths['f77'] = 'clang/flang'
-        link_paths['fc'] = 'clang/flang'
+    # Clang has support for using different fortran compilers with the
+    # clang executable.
+    @property
+    def link_paths(self):
+        # clang links are always the same
+        link_paths = {'cc': 'clang/clang',
+                      'cxx': 'clang/clang++'}
+
+        # fortran links need to look at the actual compiler names from
+        # compilers.yaml to figure out which named symlink to use
+        for string, link_path in f77_mapping:
+            if string in self.f77:
+                link_paths['f77'] = link_path
+                break
+        else:
```

Comment:
I like this idea, but I'm confused about the indentation of these `else` blocks.  What am I missing?

Reply:
Python [loop else](https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops)

Reply:
See 4.4 of https://docs.python.org/2/tutorial/controlflow.html

`for` statements can have an `else` clause which will trigger for a loop that is not terminated with a break. So the `else` is to handle the case where `link_paths['f77']` is still not set

Reply:
Better details from Nick Coughlan: http://python-notes.curiousefficiency.org/en/latest/python_concepts/break_else.html

Reply:
Well, that's neat.  Not exactly a new feature either, I wonder how I've managed to miss it.